### DISCUSSION TITLE
How Milvus work when search with expr and vector?

### DISCUSSION THREAD

**Discussion by wanglunhui2012** [milvus-io/milvus#35823] [Q&A and General discussion]
When I perform a search with expr and vector, what does Milvus filter first? by expr first, or vector?

**Comment by yhmo** [comment]
Do filtering first, and then ANN search, the filtered-out vectors will be ignored by ANN search.
It uses a bitset list to mark the vectors, 0 is filtered-out, 1 is searchable: https://milvus.io/docs/v2.3.x/bitset.md#Bitset

So, the basic logic is: filtering ---> get a bitset ---> ANN search
